app/src/test/java/com/example/habitsmasher/test/ListenersTest.java

```diff
@@ -0,0 +1,135 @@
+import com.example.habitsmasher.listeners.FailureListener;
+import com.example.habitsmasher.listeners.FailureListenerWithToast;
+import com.example.habitsmasher.listeners.SuccessListener;
+import com.example.habitsmasher.listeners.SuccessListenerWithToast;
+import com.example.habitsmasher.listeners.SwipeListener;
+import com.example.habitsmasher.ui.dashboard.HabitItemAdapter;
+import com.example.habitsmasher.ui.dashboard.HabitListFragment;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+/**
+ * This class hosts the unit tests for the listeners
+ *
+ * The reference for testing listeners is the following url:
+ * URL: https://stackoverflow.com/questions/18913827/how-to-test-listener-interface-is-called-within-android-unit-tests/19050763
+ */
+public class ListenersTest {
+    private static final String TAG = "Example tag";
+    private static final String MSG = "Example message";
+    private static final String TOAST_MSG = "Toast message";
+    private static final String DAY = "MO";
+    private static final int ID = 1;
+    private static final int POSITION = 0;
+
+    private ClickListenerForCancel _clickListenerForCancel;
+    private ClickListenerForDatePicker _clickListenerForDatePicker;
+    private ClickListenerForDaysOfTheWeek _clickListenerForDaysOfTheWeek;
+    private FailureListener _failureListener;
+    private FailureListenerWithToast _failureListenerWithToast;
+    private SuccessListener _successListener;
+    private SuccessListenerWithToast _successListenerWithToast;
+    private SwipeListener _swipeListener;
+
+    @Before
+    public void setUp() {
+        _clickListenerForCancel = Mockito.spy(new ClickListenerForCancel(mock(Dialog.class), TAG));
+        _clickListenerForDatePicker = Mockito.spy(new ClickListenerForDatePicker
+                (mock(FragmentManager.class), mock(TextView.class)));
+        _clickListenerForDaysOfTheWeek = Mockito.spy(new ClickListenerForDaysOfTheWeek
+                (mock(DaysTracker.class), DAY));
+        _failureListener = Mockito.spy(new FailureListener(TAG, MSG));
+        _failureListenerWithToast = Mockito.spy(new FailureListenerWithToast(mock(Context.class),
+                TAG, MSG, TOAST_MSG));
+        _successListener = Mockito.spy(new SuccessListener(TAG, MSG));
+        _successListenerWithToast = Mockito.spy(new SuccessListenerWithToast(mock(Context.class),
+                TAG, MSG, TOAST_MSG));
+        _swipeListener = Mockito.spy(new SwipeListener(mock(HabitItemAdapter.class),
+                mock(HabitListFragment.class), mock(HabitList.class), mock(User.class)));
+    }
+
+    @Test
+    public void clickListenerForCancel_validInput_expectNoErrors(){
+        View mockView = mock(View.class);
+        _clickListenerForCancel.onClick(mockView);
+        //Here we can verify whether the methods "onClick" executed
+        verify(_clickListenerForCancel, times(1)).onClick(mockView);
```

Comment:
If I'm reading this correctly, the `verify()` method only ensures that the method was **called X amount of times**, not on the correctness of the result.

Do we also need to assert on some expected condition, or is the fact that we received no errors good enough?